### DISCUSSION TITLE
Failed to open `*.lua` file via kitty in MacOS.

### DISCUSSION THREAD

**Discussion by Fau818** [kovidgoyal/kitty#6113] [Q&A] [answered]
I use kitty as the default program to open all `*.cpp` files, it works well.

But when I do the same thing for `*.lua` files, I get an error.

```
Open URL Error

Unknown URL type, cannot open:
  file:///Users/Fau/a.lua
Press Enter or Esc to exit
```

I searched the issues and discussions, and tried to configure `launch-actions.conf` and `open-actions.conf` files to fix the error, but failed.

So, can someone help me?

**Comment by page-down** [comment, marked as answer]
> ... tried to configure `launch-actions.conf` ... files to fix the error, but failed.

Not sure what you tried, as you didn't say.

According to the following document, and note the `ext` in the matching criteria.
https://sw.kovidgoyal.net/kitty/open_actions/#scripting-the-opening-of-files-with-kitty-on-macos

```conf
protocol file
ext lua
action ...
```

Or configure `mime.types` since there is a default `mime text/*` rule.

```conf
text/x-lua lua
```

Don't forget to reload the configuration or quit kitty.app.

**Reply by Fau818** [reply]
Thank you for your reply, it solved my problem perfectly.

---

In my previous attempt, I added the following code snippet to the `launch-actions.conf` file.

```
# Open text files without fragments in the editor
protocol file
mime text/*
action launch --type=os-window $EDITOR $FILE_PATH
```

But it doesn't work for me.

After reading your answer, I tried changing `mime text/*` to `ext lua`, and it worked.
